### PR TITLE
handle lack of tests

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

File: crates/cairo-lang-protostar/src/casm_generator/mod.rs [hunk 1]
```diff
@@ -100,6 +102,9 @@ impl SierraCasmGenerator {
             Some(result) => result,
             None => self.collect_tests().into_iter().map(|item| item.to_string()).collect(),
         };
+        if tests.is_empty() {
```
Comment (MaksymilianDemitraszek) [inline crates/cairo-lang-protostar/src/casm_generator/mod.rs:105]: Why do we need to handle it here, wouldn't be better to just return an empty list and handle the error on the side of protostar?

Comment (karol-bisztyga) [reply crates/cairo-lang-protostar/src/casm_generator/mod.rs:105]: Hmm ok, I think we can return an empty list instead of an error 👍

Comment (karol-bisztyga) [reply crates/cairo-lang-protostar/src/casm_generator/mod.rs:105]: I think we should handle this here. In this function `build_casm` we receive an argument `maybe_attributed_tests`, and if it doesn't contain test we fall back to looking for tests. The problem is, if no tests are received and none are found, we're going to end up with a nasty error "index out of range" [here](https://github.com/software-mansion-labs/cairo/pull/27/files#diff-01a9b0afdf08ecbfb6732b0853a39667fef1be639db2603b4048a2ff0b5b1354R118). This is about handling an exceptional situation for this very function not even looking at the context of the operations that are performed.

Comment (MaksymilianDemitraszek) [reply crates/cairo-lang-protostar/src/casm_generator/mod.rs:105]: Ok sounds good to me

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**